app/models/katello/repository.rb

```diff
@@ -707,14 +710,15 @@ def ensure_ostree_repo_protected
     end
 
     def remove_docker_content(manifests)
-      self.docker_tags.where(:docker_manifest_id => manifests.map(&:id)).destroy_all
-      DockerMetaTag.cleanup_tags
-      self.docker_manifests -= manifests
-
+      destroyable_manifests = manifests.select do |manifest|
+        manifest.repositories.empty? || manifest.docker_manifest_lists.empty?
+      end
       # destroy any orphan docker manifests
-      manifests.each do |manifest|
-        manifest.destroy if manifest.repositories.empty?
+      destroyable_manifests.each do |manifest|
+        self.docker_manifests.delete(manifest)
+        manifest.destroy
```

Comment:
Can't you just do `manifest.destroy` because you have this in the docker manifest model:

```rb
has_many :repository_docker_manifests, :dependent => :destroy, :inverse_of => :docker_manifest
```